### ISSUE TITLE
Concatenated Messages get merged when using BeginSendMessage, and UCS2 encoding.

### ISSUE THREAD

**Issue by MahmoodRasooli** [AdhamAwadhi/JamaaSMPP#39]
Hi Adham,
I've been using JamaaSmpp for the past nine months and lately, we've encountered a very abnormal problem. When sending large messages(more than 2 PDUs) asynchronously using BeginSendMessage, sometimes the messages get merged and the user gets only one message which is a combination of the two sent messages. The final message has the exact number of characters of one of the messages but the content is neither one of them.
P.S: It's worth mentioning that I send nearly 500 PDUs every second in rush hours.
I appreciate it if you can help me with this issue. Many thanks.

**Comment by MahmoodRasooli** [comment]
Found the problem. It was because of the SegmentId in UDH. The framework generates this number randomly, and that sometimes causes this issue, because it generates a single number every time, hence the different messages get the same SegID.

[TextMessage.cs line 83:](https://github.com/AdhamAwadhi/JamaaSMPP/blob/979b4daea325ae9f6453d1935aeba53235ca19d2/JamaaTech.Smpp.Net.Client/TextMessage.cs#L83)
`var SegID = new Random().Next(1000, 9999); // create random SegmentID`

@AdhamAwadhi

**Comment by AdhamAwadhi** [comment]
Smpp protocol give you only `1 byte` as message identification. 
[https://smpp.io/concatenated-messages/](https://smpp.io/concatenated-messages/)

If you have any solution give us a hint.

Thank you

**Comment by MahmoodRasooli** [comment]
Firstly, you did an outstanding job creating this library, well done, and my compliments to you.
Secondly, you're right, the message identification only allows 1 byte, however, I think there's a better way than generating a random number.
I used a `[Thread Static]` variable to use it as SegId(`[Thread Static]` because I need to use multiple `Transceiver` connections in my application, hence multiple `SmppClient` objects). Then I add it incrementally and if it gets bigger than `FF`, I'll reset it to 0. This way there's a better chance to at least reduce the possibility of encountering conflicts while sending messages with high speed.

I also have a question to ask you, what do you think is the best way to use your library to be able to send more than 600-700 `PDU`s per second? As you know, `Thread.Sleep()` and other workarounds like this are not acceptable.

Thanks for your time.
@AdhamAwadhi

**Comment by AdhamAwadhi** [comment]
I'm not the creator of this library. [See readme](https://github.com/AdhamAwadhi/JamaaSMPP#based-on).  I liked it and I just maintain it as possible as I can . _(Working full Time)_

Thank you for sharing your solution. Is your application a multi-threaded technology like Web applications? 

About your question: why you need something like `Thread.Sleep()`?

_And sorry for my late response. I'm working full time as I mentioned before_

**Comment by MahmoodRasooli** [comment]
Yes, the application which I’ve developed is a multi-threaded windows service(a daemon in linux).
About the Thread.Sleep(Task.Delay), the SMSC that my company is working with, limited the number of tcp connections which we can use(10 in total), and each connection has a specific TPS. So I need to adjust the application’s send rate with that throughput, otherwise I’ll get Smpp_Throttling error.
I appreciate it if you can share any other solutions if you’ve ever came across one.

Also thanks for the response, I’m a full-time developer too and I know the limitations.
